### PR TITLE
Services ready v1

### PR DESCRIPTION
**Pull request by KacperLechicki**
All services are ready for reading all data or one item by id, models are the same as in backend.

### CODE REVIEW

File: front-project/src/app/services/http/canteen.service.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  getAllFromCanteen(): Observable<Canteen[]> {
+    return this.http
+      .get<Canteen[]>(`${Constants.backendApiUrl}Main/GetAllCanteens`)
+      .pipe(retry(1), catchError(this.handleError));
+  }
+
+  getOneFromCanteen(canteenID: number): Observable<Canteen> {
+    return this.http
+      .get<Canteen>(
+        `${Constants.backendApiUrl}Main/GetCanteen/${canteenID - 1}`
```
Comment (OliwiaMilewska) [inline front-project/src/app/services/http/canteen.service.ts:22]: Why "-1"? If item in canteen is stored with id 5 in our database we should use 5 not 4 to find it.

File: front-project/src/app/services/http/categories.service.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  getAllFromCategories(): Observable<Category[]> {
+    return this.http
+      .get<Category[]>(`${Constants.backendApiUrl}Main/GetAllCategories`)
+      .pipe(retry(1), catchError(this.handleError));
+  }
+
+  getOneFromCategories(categoryID: number): Observable<Category> {
+    return this.http
+      .get<Category>(
+        `${Constants.backendApiUrl}Main/GetCategory/${categoryID - 1}`
```
Comment (OliwiaMilewska) [inline front-project/src/app/services/http/categories.service.ts:22]: Same -1?

File: front-project/src/app/services/http/consultations.service.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  getAllFromConsultations(): Observable<Consultation[]> {
+    return this.http
+      .get<Consultation[]>(`${Constants.backendApiUrl}Main/GetAllConsultations`)
+      .pipe(retry(1), catchError(this.handleError));
+  }
+
+  getOneFromConsultations(consultationID: number): Observable<Consultation> {
+    return this.http
+      .get<Consultation>(
+        `${Constants.backendApiUrl}Main/GetConsultations/${consultationID - 1}`
```
Comment (OliwiaMilewska) [inline front-project/src/app/services/http/consultations.service.ts:22]: etc.

File: front-project/src/app/models/professor.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export interface Professor {
+  id: number;
+  organisation: string;
+  name: string;
+  room: string;
+  email: string;
+  howToContact: string;
+  typeOfExamination: string;
+  availabilityOfMaterials: string;
+  note: string;
```
Comment (OliwiaMilewska) [inline front-project/src/app/models/professor.ts:10]: Missing list with consultations.

File: front-project/src/app/layouts/pages/canteen/canteen.component.ts [hunk 5]
```diff
@@ -8,11 +11,23 @@ import { LoadingService } from 'src/app/services/loading.service';
...
 export class CanteenComponent {
   loading!: boolean;
+  canteenArr: Canteen[] = [];
 
-  constructor(private loadingS: LoadingService) {}
+  getAllFromCanteenSub!: Subscription;
+
+  constructor(
+    private loadingS: LoadingService,
+    private canteenS: CanteenService
```
Comment (OliwiaMilewska) [inline front-project/src/app/layouts/pages/canteen/canteen.component.ts:20]: To be honest maybe we can change 'canteenS' to 'canteenService'? And the same in others .ts files.
image

File: front-project/src/app/layouts/pages/canteen/canteen.component.ts [hunk 6]
```diff
@@ -8,11 +11,23 @@ import { LoadingService } from 'src/app/services/loading.service';
 })
 export class CanteenComponent {
   loading!: boolean;
+  canteenArr: Canteen[] = [];
```
Comment (OliwiaMilewska) [inline front-project/src/app/layouts/pages/canteen/canteen.component.ts:14]: It is good practice that when we see that variable is list, array, table etc. then we don't call it nameArray, nameTable. This list will contain all dishes from our canteen so maybe we can call it more meaningful? For example dishes: Canteen[].

File: front-project/src/app/layouts/pages/dictionary/dictionary.component.ts [hunk 7]
```diff
@@ -1,18 +1,33 @@
...
 
 @Component({
   selector: 'app-dictionary',
   templateUrl: './dictionary.component.html',
-  styleUrls: ['./dictionary.component.css']
+  styleUrls: ['./dictionary.component.css'],
 })
 export class DictionaryComponent {
   loading!: boolean;
+  dictionaryArr: Dictionary[] = [];
```
Comment (OliwiaMilewska) [inline front-project/src/app/layouts/pages/dictionary/dictionary.component.ts:14]: Give more meaningful name.

File: front-project/src/app/layouts/pages/dictionary/dictionary.component.ts [hunk 8]
```diff
@@ -1,18 +1,33 @@
...
 
-  constructor(private loadingS: LoadingService) {}
+  getAllFromDictionarySub!: Subscription;
+
+  constructor(
+    private loadingS: LoadingService,
+    private dictionaryS: DictionaryService
```
Comment (OliwiaMilewska) [inline front-project/src/app/layouts/pages/dictionary/dictionary.component.ts:20]: Change naming conventions for variables relevant to services.

File: front-project/src/app/layouts/pages/forum/forum.component.ts [hunk 9]
```diff
@@ -8,11 +15,41 @@ import { LoadingService } from 'src/app/services/loading.service';
 })
 export class ForumComponent {
   loading!: boolean;
+  postsArr: Post[] = [];
```
Comment (OliwiaMilewska) [inline front-project/src/app/layouts/pages/forum/forum.component.ts:18]: If here we have array of posts we can just change it from 'postsArr' to 'posts'.

File: front-project/src/app/layouts/pages/forum/forum.component.ts [hunk 10]
```diff
@@ -8,11 +15,41 @@ import { LoadingService } from 'src/app/services/loading.service';
...
+  categoriesArr: Category[] = [];
 
-  constructor(private loadingS: LoadingService) {}
+  getAllFromPostsSub!: Subscription;
+  getAllFromRepliesSub!: Subscription;
+  getAllFromCategoriesSub!: Subscription;
+
+  constructor(
+    private loadingS: LoadingService,
+    private postsS: PostsService,
```
Comment (OliwiaMilewska) [inline front-project/src/app/layouts/pages/forum/forum.component.ts:28]: I think you already know what to do :)

**Review by OliwiaMilewska — CHANGES_REQUESTED** [review]

File: front-project/src/app/services/http/canteen.service.ts (hunk 1 above)
Comment (KacperLechicki) [reply front-project/src/app/services/http/canteen.service.ts:22]: Yes, but it is working funny and when you give this method id = 0, then you get item with id = 1 from database. So, if you give id = 1, you get id = 2 from database, etc.

**Review by OliwiaMilewska — APPROVED** [review]

**Pull request merged**